ui/tests/acceptance/exec-test.js

```diff
@@ -276,6 +278,44 @@ module('Acceptance | exec', function(hooks) {
     );
   });
 
+  test('the opening message includes the token if it exists', async function(assert) {
+    const { secretId } = server.create('token');
+    window.localStorage.nomadTokenSecret = secretId;
+
+    let mockSocket = new MockSocket();
+    let mockSockets = Service.extend({
+      getTaskStateSocket() {
+        return mockSocket;
+      },
+    });
+
+    this.owner.register('service:sockets', mockSockets);
+
+    let taskGroup = this.job.task_groups.models[0];
+    let task = taskGroup.tasks.models[0];
+    let allocations = this.server.db.allocations.where({
+      jobId: this.job.id,
+      taskGroup: taskGroup.name,
+    });
+    let allocation = allocations[allocations.length - 1];
+
+    await Exec.visitTask({
+      job: this.job.id,
+      task_group: taskGroup.name,
+      task_name: task.name,
+      allocation: allocation.id.split('-')[0],
+    });
+
+    await Exec.terminal.pressEnter();
+    await settled();
+    mockSocket.onopen();
+
+    await Exec.terminal.pressEnter();
+    await settled();
+
+    assert.equal(mockSocket.sent[0], `{"version":1,"auth_token":"${secretId}"}`);
+  });
```

Comment:
I pared down this test somewhat to remove assertions that were already covered in the preceding test, so now it’s exclusively focused on verifying that the token is sent as expected.